### PR TITLE
Add script to compile renderer against TOT devtools frontend

### LINKED ISSUE #2050: Compile renderer against DevTools frontend
Now with brendan's new compilation in place, we should also be able to check that we compile against the devtools frontend, which has slightly different compilation settings.

Basically i'm thinking we'll use latest devtools frontend, copy our renderer sources in there, and run its closure.

-------------

We have a few prerequisites:
1. paul lands the ReportRenderer CL in devtools
1. will adds a hook to compile_frontend.py to provide a custom path for protocol_externs
1. paul adds a hook to our roll-to-devtools.sh to provide a different `$frontend_dir` location.

-----------

Then I think the flow looks something like this

Add latest devtools frontend to devDeps:
```
 "devtools-frontend-tot": "git://github.com/ChromeDevTools/devtools-frontend.git",
```
Probably add [ChromeDevTools/devtools-protocol) to devDeps as well.

Run roll-to-devtools.sh with `./node_modules/devtools-frontend-tot/front_end`, which copies our renderer files over.

Reach into the frontend and run closure:
```sh
 npm explore devtools-frontend-tot -- npm run closure
```

### PR DESCRIPTION
**Pull request by paulirish**
fixes #2050 

output:

image

### CODE REVIEW

File: lighthouse-core/scripts/compile-against-devtools.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+if [ ! -d "$protocol_path" ]; then
+  git clone --depth=1 git://github.com/ChromeDevTools/devtools-protocol.git "$protocol_path"
+fi
+
+# update to latest
+cd "$frontend_path" && git reset --hard && git fetch origin master && git checkout --force origin/master
+cd "$protocol_path" && git reset --hard && git fetch origin master && git checkout --force origin/master
+
+
+cd "$lhroot_path" || exit 1
```
Comment (patrickhulce) [inline lighthouse-core/scripts/compile-against-devtools.sh]: does this fail sometimes?

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "unit": "bash lighthouse-core/scripts/run-mocha.sh --default",
     "closure": "cd lighthouse-core && node closure/closure-type-checking.js",
     "devtools": "bash lighthouse-core/scripts/roll-to-devtools.sh",
+    "compdevtools": "bash lighthouse-core/scripts/compile-against-devtools.sh",
```
Comment (patrickhulce) [inline package.json]: I know you love 🖐 but I didn't know what this meant haha, can we do `compile-devtools`? it's only 4 more characters :)

Comment (paulirish) [inline package.json]: sure sure. 
i found and installed `https://github.com/mklabs/yarn-completions` which does completions for our scripts on `yarn run` so i'm happy now. :)

File: lighthouse-core/scripts/compile-against-devtools.sh (hunk 1 above)
Comment (paulirish) [inline lighthouse-core/scripts/compile-against-devtools.sh]: no but https://www.shellcheck.net/ recommends it so i do it. :)

**Comment by paulirish** [comment]
@brendankenny merge when ready

**Comment by paulirish** [comment]
> should we add this to CONTIBUTING?

sure. 

> should we add this to testing?

the plan is to flip it on once we get it passing on master. (last i checked master wasn't yet passing)

> After a perusal it appears this is non-destructive on my Lighthouse repo, so I can run multiple times/run in an in-progress change.

totally. run as much as you like. doesn't affect your LH files.

**Comment by brendankenny** [comment]
> last i checked master wasn't yet passing

yeah, that's what I just ran into.  LGTM!

**Comment by paulirish** [comment]
added a note to readme:
image

**Pull request merged**